lib/Runtime/Debug/DebugContext.h

```diff
@@ -51,10 +51,17 @@ namespace Js
         void SetHostDebugContext(HostDebugContext * hostDebugContext);
 
         void SetDebuggerMode(DebuggerMode mode);
+        void SuspendRecording();
+        void ResumeRecording();
+
+        DebuggerMode GetDebuggerMode() { return this->debuggerMode; }
         bool IsDebugContextInNonDebugMode() const { return this->debuggerMode == DebuggerMode::NotDebugging; }
-        bool IsDebugContextInDebugMode() const { return this->debuggerMode == DebuggerMode::Debugging; }
+        bool IsDebugContextInDebugMode() const { return debuggerMode == DebuggerMode::Debugging; }
         bool IsDebugContextInSourceRundownMode() const { return this->debuggerMode == DebuggerMode::SourceRundown; }
         bool IsDebugContextInSourceRundownOrDebugMode() const { return IsDebugContextInSourceRundownMode() || IsDebugContextInDebugMode(); }
+        bool IsDebugContextInDebugRecordingMode() const { return this->IsDebugContextInDebugMode() && this->IsDebuggerRecording(); }
+        bool IsDebugContextInDebugRecordingSuspendedMode() const { return this->IsDebugContextInDebugMode() && !this->IsDebuggerRecording(); }
```

Comment:
I don't think there is any need for these 2 functions. 
IsDebuggerRecording should have an assert that we are debugging and done.